test/parallel/test-stream-write-destroy.js

```diff
@@ -34,14 +34,14 @@ for (const withPendingData of [ false, true ]) {
     if (withPendingData) {
       // Test 2 cases: There either is or is not data still in the write queue.
       // (The second write will never actually get executed either way.)
-      w.write('def', () => chunksWritten++);
+      w.write('def', (err) => !err && chunksWritten++);
     }
     if (useEnd) {
       // Again, test 2 cases: Either we indicate that we want to end the
       // writable or not.
-      w.end('ghi', () => chunksWritten++);
+      w.end('ghi', (err) => !err && chunksWritten++);
     } else {
-      w.write('ghi', () => chunksWritten++);
+      w.write('ghi', (err) => !err && chunksWritten++);
```

Comment:
Would you mind adding some assertion when `err` is set in all those changes?

Reply:
I'm not sure I understand?

Reply:
I guess you had to add that `!err &&` because the test was failing otherwise. If we expect errors to happen, we should add some assertions so that's codified behavior and we do not regress in the future.

Reply:
test updated